### ISSUE TITLE
With python3.8 date_time_between_dates returns faker.utils.datetime_safe.datetime instead of datetime.datetime

### ISSUE THREAD

**Issue by Rafalkufel** [joke2k/faker#1201]
* Faker version: 4.1.0
* OS: Linux
* Python3.8

### Brief summary of the issue goes here.

`date_time_between_dates` return `faker.utils.datetime_safe.datetime` instead of `datetime.datetime`
Problem stem from solution of python bug: https://bugs.python.org/issue32417
Faker works fine with earlier version of python (eg 3.7)

### Steps to reproduce
```
from datetime import datetime, timedelta

from faker import Faker

fake = Faker()
Faker.seed(0)

fake_date = fake.date_time_between_dates(
    datetime_start=datetime.utcnow() - timedelta(days=360),
    datetime_end=datetime.utcnow(),
    tzinfo=None
)
    
print(type(fake_date))
```
### Expected behavior
output:
<class 'datetime.datetime'>

### Actual behavior
output:
<class 'faker.utils.datetime_safe.datetime'>

**Comment by fcurella** [comment]
Hi! Thank you for reporting this.

My understanding from the Python bug is that returning a `datetime.datetime` was a bug in the first place, and returning a `datetime_safe.datetime` is what's considered expected behavior by Python.

Could be more specific about what's breaking for you?

**Comment by Rafalkufel** [comment]
Sorry, maybe I didn't make myself clear.

Using `date_time_between_dates` provider I would expect that the fake object which returns is the same class as the real one (eg `datetime.datetime` when it comes to date). And that is the way it works with previous version of python. But surprinsingly, with python3.8 it returns object of different type than expected. In my case `pymysql` starts receiving `faker.utils.datetime_safe.datetime` object in place where `datetime.datetime` is expected.

I think that change in the way python3.8 datetime works should be reflected also in Faker to get real `datetime` from datetime provider.

**Comment by samcrang** [comment]
We're also seeing a problem with this.

I'd expect the `faker.date*()` methods to return a regular Python `date`/`datetime` objects not the subtyped `date`/`datetime`s Faker uses internally. Since Python3.8 we're seeing the latter behaviour.

Regular `date`/`datetime` objects are returned when running on Python3.7 however it appears [the behaviour of arithmetic on subtyped `date`/`datetime` objects changed in Python3.8](https://github.com/python/cpython/commit/d9503c307a8b6a7b73f6344183602ffb014d3356).

On Python 3.7 we get this:

```
Python 3.7.8 (default, Sep  2 2020, 22:20:06)
[Clang 11.0.0 (clang-1100.0.33.17)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from faker import Faker
>>> f = Faker()
>>> f.past_date().__class__
<class 'datetime.date'>
```

On Python 3.8 we get:

```
Python 3.8.5 (default, Aug 18 2020, 15:15:26)
[Clang 11.0.0 (clang-1100.0.33.17)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from faker import Faker
>>> f = Faker()
>>> f.past_date().__class__
<class 'faker.utils.datetime_safe.date'>
```

Like @Rafalkufel we see issues when these subtyped `datetime` are used by `pymysql`.

For what it's worth we're currently working around this by doing something like this in our code:

```
def _fix_faker_datetimes_on_python38(dt):
  if dt.__class__.__module__ == 'faker.utils.datetime_safe':
    return datetime.fromisoformat(dt.isoformat())
  return dt
```

**Comment by fcurella** [comment]
I'm not sure how to make sure datetime operations return a `datetime` object. Any idea on how this could be fixed?

**Comment by samcrang** [comment]
My understanding is that Faker only supports Python >= 3.5 (from the `.travis.yml`).

If that's the case then I think it should be acceptable to remove [the workaround which subtypes `datetime`](https://github.com/joke2k/faker/blob/master/faker/utils/datetime_safe.py). According to the comment it's only needed to support `strftime` on dates before 1900. From my testing it appears that's only a problem for older versions of Python:

Python 2.7:

```
Python 2.7.17 (default, Dec 23 2019, 21:25:34)
[GCC 4.2.1 Compatible Apple LLVM 11.0.0 (clang-1100.0.33.16)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import datetime
>>> datetime.datetime(1800, 12, 11)
datetime.datetime(1800, 12, 11, 0, 0)
>>> datetime.datetime(1800, 12, 11).strftime('%Y-%m-%d')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: year=1800 is before 1900; the datetime strftime() methods require year >= 1900
```

Python 3.5

```
Python 3.5.9 (default, Sep 10 2020, 22:25:21)
[GCC 4.2.1 Compatible Apple LLVM 11.0.0 (clang-1100.0.33.17)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import datetime
>>> datetime.datetime(1800, 12, 11)
datetime.datetime(1800, 12, 11, 0, 0)
>>> datetime.datetime(1800, 12, 11).strftime('%Y-%m-%d')
'1800-12-11'
```
I think PR #1258 should fix this.

**Comment by fcurella** [comment]
Closed in #1258
